### PR TITLE
Add fs module support

### PR DESCRIPTION
**Pull request by luismiramirez**
All fs module usage is now automatically instrumented.

Tests are in the express-redis sample application.

Automatic instrumentation is manually suppresed in the test span processor because it uses the fs module to handle the spans.json file. It is done using the `supressTracing` helper provided by OpenTelemetry core library.

Part of #763.

### CODE REVIEW

**Review by jeffkreeftmeijer — APPROVED** [review]

**Comment by unflxw** [comment]
I have looked into the issue where the Prisma test applications fail to boot when the FS instrumentation is added.

TL;DR: We gotta wait for a release containing [the changes in this PR](https://github.com/open-telemetry/opentelemetry-js-contrib/pull/1222) to be published (should be included in [the next release](https://github.com/open-telemetry/opentelemetry-js-contrib/pull/1214)) and update the `@opentelemetry/instrumentation-fs` package to that version.

The core of the issue is that the Prisma client calls `util.promisify(fs.exists)`. It turns out that `fs.exists` is a bit unusual. It's deprecated by the Node team, because its callback signature does not follow the standard `(err, result) => { ... }` signature of Node callbacks. Instead, since `fs.exists` can't fail, its signature is just `(result) => { ... }`.

Since `util.promisify` expects the first argument of the callback to be an error, this means that promisifying `fs.exists` would result on the promise always being rejected, with the actual result being the rejection error.

This is consistent with the error message we see:
> ```
> error2.clientVersion = this.client._clientVersion;
> ```
> ```
> TypeError: Cannot create property 'clientVersion' on boolean 'true'
> ```

Prisma is trying to set the clientVersion property on the error object it received, but the error object isn't an object, but a boolean `true`, the return value of `fs.exists`, turned into the rejection value of a promise by `util.promisify`.

Now, the reason why this bug only happens for the instrumented version of the function is that the original `fs.exists` provides a custom promisified implementation of itself. If there is an `util.promisify.custom` symbol as a property of the function, `util.promisify` will use its value as the promisified implementation. The original `fs.exists` provides this, but the OpenTelemetry-instrumented version of the function doesn't.

Or rather, it doesn't _yet_, as there's [a merged PR implementing this custom behaviour](https://github.com/open-telemetry/opentelemetry-js-contrib/pull/1222) for the instrumented version of the function. This should fix our problems; we just need to wait for [a release including the change](https://github.com/open-telemetry/opentelemetry-js-contrib/pull/1214).

I have not been able to verify the actual fix directly (npm's "install from GitHub" does not play well with monorepos) but I have confirmed that manually excluding `fs.exists` from being instrumented by commenting the relevant line of code fixes the issue, so I expect the fix above to also fix the issue.

**Review by unflxw — APPROVED** [review]
